### PR TITLE
chore: Fixes `search_deployment` template to fix doc structure

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

[Terraform registry docs](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs) for mongodbatlas are displayed with a wrong structure since v1.15.0 was released. The issue can be seen in the following image:

Screenshot 2024-02-14 at 12 16 42

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Comment by oarbusi** [comment]
[Subcategories](https://developer.hashicorp.com/terraform/registry/providers/docs#subcategories) were not used in any resource before using it in this resource. Changing it back to avoid wrong docs structure issues.

File: templates/resources/search_deployment.md.tmpl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 layout: "mongodbatlas"
 page_title: "MongoDB Atlas: {{.Name}}"
-subcategory: "docs_{{ .Name }}_{{.Type | lower}}"
+sidebar_current: "docs-{{ .ProviderShortName }}-{{ .Type | lower }}-{{ $name := split .Name "_" }}{{ index $name 1 }}-{{ index $name 2 }}"
```
Comment (andreaangiolillo) [inline templates/resources/search_deployment.md.tmpl]: Could you also update the templates https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/templates/data-source.md.tmpl and https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/templates/resources.md.tmpl?

**Review by andreaangiolillo — COMMENTED** [review]
one small comment. Great catch 💯

**Review by lantoli — APPROVED** [review]

File: templates/resources/search_deployment.md.tmpl (hunk 1 above)
Comment (oarbusi) [reply templates/resources/search_deployment.md.tmpl]: Good point, I updated the 2 templates. I also changed a bit how it works so it's more generic and would work for all resources even if the resource name has a different amount of words than search_deployment

**Review by andreaangiolillo — APPROVED** [review]
LGTM. Thanks for all the changes here

**Comment by lantoli** [comment]
I don't know if the link in make website can help to preview the changes here: https://registry.terraform.io/tools/doc-preview

**Comment by andreaangiolillo** [comment]
> I don't know if the link in make website can help to preview the changes here: https://registry.terraform.io/tools/doc-preview

Thanks for sharing the link! I am adding it to our CONTRIBUTING.md in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1945

**Comment by oarbusi** [comment]
> I don't know if the link in make website can help to preview the changes here: https://registry.terraform.io/tools/doc-preview

@lantoli tried it but it doesn't allow to preview the structure of all the docs, which in this case was the issue

**Pull request merged**